### PR TITLE
Updated README, initial revision of reference manual

### PR DESCRIPTION
**Pull request by nblair**
Also isolated release process to separate file.

### CODE REVIEW

**Comment by apetro** [comment]
wrt the file naming pattern of `README-thing.md`.  Consider instead `docs/releasing.md` and `docs/reference.md` ?

**Comment by timlevett** [comment]
Could also use wiki?

**Comment by nblair** [comment]
I'm amenable to either change.

**Comment by nblair** [comment]
Should I just move both releasing and reference to the wiki?

**Comment by apetro** [comment]
I'm :-1: on moving documentation to the wiki.  Keeping the documentation in the Git repo, and in the same branch as the code, allows versioning documentation with the code, and updating the documentation in the same changesets that update the code being documented.

**Comment by timlevett** [comment]
Sounds good @apetro . docs works for me too.

File: README.md [hunk 1]
```diff
@@ -5,33 +5,69 @@ A Simple server side REST proxy service written in Groovy
...
- * Picture this module deployed as a façade for ESB endpoints. `my.wisc.edu/esb/someservice/api/foo` . You could theoretically write 100% javascript apps against the proxy proxying ESB-provided JSON.
+*But what if the web application hosting the REST API doesn't support CORS?* That's where rest-proxy can help.
 
-### Add Proxy Servlet to Existing Service
-+ A Spring `@Configuration` class is provided, simply `@Import(edu.wisc.my.restproxy.config.RestProxyConfiguration)` on one of your existing `@Configuration` classes. If you are using Spring's XML configuration instead, add the following to your `Spring application context`:
+Without rest-proxy, imagine how you would have to integrate these two applications:
+
+1. The application hosting the REST API will likely have separate authentication mechanism, say a "service account" using HTTP Basic/Digest.
+2. In order for your web application to make REST calls, your application is going to have to have it's own REST Controllers which are simply a Proxy for the other application. You'll need a proxy method/controller for every method on the target REST API you want to call.
```
Comment (apetro) [inline README.md]: "it's" --> "its"

File: README.md [hunk 2]
```diff
@@ -5,33 +5,69 @@ A Simple server side REST proxy service written in Groovy
...
     </servlet>
     <servlet-mapping>
         <servlet-name>rest-proxy-api</servlet-name>
-        <url-pattern>/api/*</url-pattern>
+        <url-pattern>/proxy/*</url-pattern>
     </servlet-mapping>
-    
 ```
-+ Verify it works
+4. Now we have to configure the proxy. rest-proxy gets it's configuration about the target REST APIs you want to proxy from the Spring `@Environment`. If you have a `@PropertySource` already defined, you can add the necessary properties there, or you can choose to add a separate `@PropertySource` (it doesn't matter, they all end up in `@Environment` no matter how many `@PropertySource`s you have). The most basic configuration available is as follows:
```
Comment (apetro) [inline README.md]: "it's" --> "its"

**Comment by apetro** [comment]
:+1:

**Comment by timlevett** [comment]
:+1:

**Pull request merged**